### ISSUE TITLE
Download CSV option in Saved Search

### ISSUE THREAD

**Issue by Sandy381** [opensearch-project/dashboards-reports#378]

**Comment by Sandy381** [comment]
@joshuarrrr @kavilla

We can download a saved search as CSV , but only in discover.
We cannot download a saved search in a dashboard & that is what is required.

**Comment by zhongnansu** [comment]
Same feature request as https://github.com/opensearch-project/dashboards-reports/issues/175 . This is a highly requested feature, and it has been brought up quite often.

**Comment by mbonnici55** [comment]
Would this be an appropriate place to ask if we could just have the 'download' as a CSV even without a saved search? For people that just want to use the 'Discover' tab and filter down to data they would like to see, then export it and do further analysis on it. It seems that just being allowed to 'download' your current search as a CSV to share would be helpful.

**Comment by joshuarrrr** [comment]
Yeah, @mbonnici55, I think we may update the table portion of discover to a component that has native support for CSV exports. @ananzh What do you think - could this be in scope for discover as part of using `DataGrid`?
